rs/sns/governance/src/types.rs

```diff
@@ -2160,6 +2176,250 @@ impl NeuronParameters {
+                            number of permissions ({})",
+                        permissions.len(),
+                        max_number_of_principals_per_neuron
+                    ));
+                }
+            }
+            Err(e) => {
+                defects.push(e);
+            }
+        }
+
+        if !defects.is_empty() {
+            let participant_info = match participant {
+                Some(Participant::Direct(_)) => {
+                    format!("direct participant {:?}", self.controller)
+                }
+                Some(Participant::NeuronsFund(nf)) => {
+                    format!("neurons fund participant {:?}", nf.nns_neuron_id)
+                }
+                None => "unknown participant".to_string(),
+            };
+
+            Err(format!(
+                "Could not claim neuron for {} with NeuronId {:?} due to: {}",
+                participant_info,
+                neuron_id,
+                defects.join("\n"),
+            ))
+        } else {
+            Ok(())
+        }
+    }
+
+    pub(crate) fn is_community_fund_neuron(&self) -> bool {
+        matches!(self.participant, Some(Participant::NeuronsFund(_)))
+    }
+
+    #[track_caller]
+    pub(crate) fn get_dissolve_delay_seconds_or_panic(&self) -> u64 {
+        self.dissolve_delay_seconds
+            .expect("Expected the dissolve_delay_seconds to be present in NeuronRecipe")
+    }
+
+    #[track_caller]
+    pub(crate) fn get_stake_e8s_or_panic(&self) -> u64 {
+        self.stake_e8s
+            .expect("Expected the stake_e8s to be present in NeuronRecipe")
+    }
+
+    #[track_caller]
+    pub(crate) fn get_neuron_id_or_panic(&self) -> &NeuronId {
+        self.neuron_id
+            .as_ref()
+            .expect("Expected NeuronId to be present in NeuronRecipe")
+    }
+
+    pub(crate) fn source_nns_neuron_id(&self) -> Option<u64> {
+        match &self.participant {
+            Some(Participant::NeuronsFund(nf)) => nf.nns_neuron_id.as_ref().cloned(),
+            _ => None,
+        }
+    }
+
+    #[track_caller]
+    pub(crate) fn construct_permissions_or_panic(
+        &self,
+        neuron_claimer_permissions: NeuronPermissionList,
+    ) -> Vec<NeuronPermission> {
+        self.construct_permissions(neuron_claimer_permissions)
+            .expect("Failed to construct permissions for neuron")
+    }
+
+    pub(crate) fn construct_permissions(
+        &self,
+        neuron_claimer_permissions: NeuronPermissionList,
+    ) -> Result<Vec<NeuronPermission>, String> {
+        let mut permissions = vec![];
+
+        let controller = self
+            .controller
+            .as_ref()
+            .ok_or("Expected controller to be present in NeuronRecipe".to_string())?;
+
+        permissions.push(NeuronPermission::new(
+            controller,
+            neuron_claimer_permissions.permissions,
+        ));
+
+        match &self.participant {
```

Comment:
This match statement does two things:

1. It checks that `self.participant` is set
2. It adds additional permissions for NF participants.

Could you split this into two statements that do only one thing each, please? I'd suggest first validating condition 1., then having a simple if-statement for 2.

Reply:
Done